### PR TITLE
lame: Add [provides] section

### PR DESCRIPTION
**Pull request by thiblahute**
See https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/1068#note_1095243

### CODE REVIEW

**Comment by thiblahute** [comment]
CI should be fixed, not ideal as we have our own `.h` file, but it works.

**Comment by eli-schwartz** [comment]
Not only is it not ideal, the original rationale for it was mistaken.

The correct solution there was always to use vs_module_defs. Additionally it is possible to support symbol visibility on GCC by reimplementing libtool's version map: https://github.com/mesonbuild/meson/issues/3047

I hacked it. :)

**Comment by eli-schwartz** [comment]
Weird: https://github.com/mesonbuild/wrapdb/runs/3878392317#step:6:232

The vs module defs file doesn't seem to work?

**Comment by thiblahute** [comment]
> The vs module defs file doesn't seem to work?

Looks like it.

Using symbol visibility prototypes makes it simpler to me tbh.

File: tools/sanity_checks.py [hunk 1]
```diff
@@ -38,6 +38,9 @@
         'meson.build.tmpl',
         'README.md',
     ],
+    'lame': [
+        'include/lame.h',
```
Comment (xclaesse) [inline tools/sanity_checks.py:42]: I think it's only the file name, not the path.

**Comment by eli-schwartz** [comment]
> Using symbol visibility prototypes makes it simpler to me tbh.

Not forking the upstream header makes it simpler to me!

**Comment by xclaesse** [comment]
> While I don't disagree, it is orthogonal to that PR :-)

while I don't disagree, we are trying to take the opportunity of wrap updates to improve them, otherwise nobody will... :P

**Pull request merged**